src/lib.rs

```diff
@@ -26,10 +25,6 @@ mod main_macro;
 mod reexports {
     #[doc(hidden)] pub use serde_derive::*;
 
-    #[doc(hidden)] pub use structopt_derive::*;
-    #[doc(hidden)] pub mod structopt {
-        pub use ::structopt::*;
-    }
```

Comment:
I think you still need something like that. At worst, it should work as before if you don't remove all the section.

Note that the custom derive will generate something containing ```::structopt::StructOpt``` thus tour reexport dance should include something like that.